### PR TITLE
Happy-thoughts-api-Ekaterina

### PR DESCRIPTION
**Pull request by klimenko3007**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,28 +1,93 @@
...
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+    trim: true,
+    validate: {
+      validator: (value) => {
+        return !(/\b(fuck)\b/i.test(value))
+      },
+      message: "Cursing words are not allowed"
```
Comment (rubmaigut) [inline server.js:21]: Nice validator "Only positive thoughts"

File: server.js [hunk 2]
```diff
@@ -1,28 +1,93 @@
...
+      {
+        $skip: ((+page - 1) * perPage) || 0
+      },
+      {
+        $limit: +perPage || 20
+      }
+    ])
+    res.json(thoughts)
+  } catch (error) {
+    res.status(400).json(error)
```
Comment (rubmaigut) [inline server.js:87]: Ohh good way to make pagination, you inspire me !!

**Review by rubmaigut — APPROVED** [review]
Good code, clear, and easy to read. !

**Review by rubmaigut — APPROVED** [review]
Good code, clear, and easy to read. !

**Comment by rubmaigut** [comment]
Good work!

File: server.js (hunk 1 above)
Comment (KarolCislowski) [inline server.js:21]: Nicely configured field 🤓

File: server.js [hunk 3]
```diff
@@ -1,28 +1,93 @@
...
+      {
+        $sort: {
+          createdAt: -1
+        }
+      },
+      {
+        $skip: ((+page - 1) * perPage) || 0
+      },
+      {
+        $limit: +perPage || 20
```
Comment (KarolCislowski) [inline server.js:82]: awesome 😻

File: server.js [hunk 4]
```diff
@@ -1,28 +1,93 @@
...
+    await thought.save()
+    res.json(thought)
+  } catch (error) {
+    res.status(400).json({ error })
+  }
+})
+app.patch('/thoughts/:id/hearts', async (req, res) => {
+  const { id } = req.params
+  try {
+    const thoughtUpdate = await Thought.findByIdAndUpdate(id, { $inc: { hearts: 1 } }, { new: true })
```
Comment (KarolCislowski) [inline server.js:59]: 💯

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Your code looks great but you need to provide a link to the frontend connected to this API.

**Review by KarolCislowski — APPROVED** [review]
Great APP!

**Pull request closed without merging**